hknweb/events/views.py

```diff
@@ -23,6 +25,39 @@ def index(request):
     }
     return render(request, 'events/index.html', context)
 
+@method_login_and_permission('events.add_rsvp')
+class AllRsvpsView(TemplateView):
+    """ List of rsvp'd and not rsvp'd events. """
+    template_name = 'events/all_rsvps.html'
+
+    def get_context_data(self):
+        view_option = self.request.GET.get('option')
+        semester_start, semester_end = get_semester_bounds(timezone.now())
+        all_events = Event.objects \
+                .filter(start_time__gte=semester_start) \
+                .filter(start_time__lte=semester_end) \
+                .order_by('start_time')
+        if view_option == "upcoming":
+            all_events = all_events.filter(start_time__gte=timezone.now())
+        rsvpd_event_ids = Rsvp.objects \
+                .filter(user__exact=self.request.user) \
+                .values_list('event', flat=True)
+        rsvpd_events = all_events \
+                .filter(pk__in=rsvpd_event_ids)
+        not_rsvpd_events = all_events \
+                .exclude(pk__in=rsvpd_event_ids)
+
+        for event in rsvpd_events:
+            event.waitlisted = event.on_waitlist(self.request.user) # Is this bad practice? idk
```

Comment:
Why is this necessarily a bad thing?

Reply:
The thing is, the waitlist is constantly changing when people RSVP, un-RSVP, or change the limit. Having multiple sources of truth is a gateway for bugs. We could make the only source of truth to make users stored as part of events instead of having RSVP objects but I don't think that's necessarily better

Reply:
I think I misunderstood the code and I am wrong! Resolving.